lib/src/cli/archive/archive_command.dart

```diff
@@ -47,12 +47,10 @@ class ArchiveCommand extends Command<void> {
       abort("source-dir option not specified");
     }
 
-    if (options.outputFile == null) {
-      abort("output-file option not specified");
-    }
+    final outputFile = options.outputFile ?? '${options.sourceDir}.tar.gz';
```

Comment:
Output file is marked as required. So we should be validating it here as before. Or if we really want to make it optional than we need to change `options.dart`. Imo being explicit here only helps, cause this command is an internal one and not something users will be using. So being explicit here only helps us - where goes what when reading the various cmake/gh actions scripts.